### PR TITLE
Adding small and right inputs, fixing frame height on Choice.

### PR DESCRIPTION
**Pull request by 1Copenut**
### Added
* CSS for proper handling of checked children with small inputs
* CSS for checked children with right to left order
* `<Choice />` component examples to support the previous two items
* Documentation for using these examples

### Changed
* Fixed the choice frame example by setting all children open with `defaultChecked`

### Design considerations
I made a few judgment calls on the inputs to the right of labels, with regard to padding and margin. Please advise if these need tweaked. The full-width didn't feel right, so I capped width, and sight-adjusted, dipping back into UI design for a moment.

### Related Issues
* Closes https://github.com/CMSgov/design-system/issues/333
* Closes https://github.com/CMSgov/design-system/issues/337

---
screen shot 2019-02-21 at 9 03 59 am

### CODE REVIEW

**Comment by line47** [comment]
I'm wondering if we should have that`Right checkboxes with children` example. That seems like a pattern we would not want to promote and maybe even discourage. 

Taking that one step further, the should we have a `Right-to-Left` option for checkboxes at all? Again this seems like it makes the checkboxes harder to use.

**Comment by 1Copenut** [comment]
@line47 I was wondering that myself. I'm not thrilled with that inputs to the right pattern, but feel we need to account for it, given it's been in the design system for some time, maybe since the beginning.

Now, we could recommend users **not** include `checkedChildren` when using that pattern and remove the styling overrides for it, but I think that has risk too.

**Comment by melwoodard** [comment]
Agreed with @line47. At minimum I'd prefer to exclude the checked children functionality from the `Right-to-Left` checkbox pattern. We should also discuss (in a new issue) whether the `Right-to-Left` pattern is something we want to include at all, and what steps we may need to take to properly deprecate it in case it's being used somewhere.

**Comment by 1Copenut** [comment]
Issue's posted for discussing deprecation. References this PR for context. I'll update the PR and docs later today.

File: packages/core/src/components/ChoiceList/Choice.scss [hunk 1]
```diff
@@ -98,9 +98,29 @@ $ds-c-inset-border-width: $spacer-half;
     margin-left: ($choice-size / 2) - ($ds-c-inset-border-width / 2);
     padding: $spacer-2;
 
+    // Checked children container on dark background
     &--inverse {
       border-left-color: $color-white;
     }
+
+    // Inputs to the right of labels
```
Comment (line47) [inline packages/core/src/components/ChoiceList/Choice.scss]: Recommend removing styling for right aligned checked children

File: packages/core/src/components/ChoiceList/Choice.scss [hunk 2]
```diff
@@ -98,9 +98,29 @@ $ds-c-inset-border-width: $spacer-half;
...
+    &--right {
+      border-left: 0;
+      border-right: $ds-c-inset-border-width solid $color-primary;
+      margin-left: 0;
+      margin-right: ($choice-size / 2) - ($ds-c-inset-border-width / 2);
+      padding: $spacer-1 0 $spacer-1 ($spacer-2 * 0.75);
+    }
+
+    // Max recommended parent width is $ds-u-measure--narrow (12em)
```
Comment (line47) [inline packages/core/src/components/ChoiceList/Choice.scss]: Recommend removing styling for width on right aligned checked children

File: packages/core/src/components/ChoiceList/Choice.example.jsx [hunk 3]
```diff
@@ -94,15 +93,74 @@ ReactDOM.render(
...
+        size="small"
+      >
+        Checkbox B - with children
+      </Choice>
+      <Choice name="checkbox_choice_children_small" size="small" value="c">
+        Checkbox C
+      </Choice>
+    </fieldset>
+
+    <div className="ds-u-measure--narrow ds-u-margin-top--4 ds-u-margin-bottom--5 ds-u-padding--2 ds-u-border--2">
```
Comment (line47) [inline packages/core/src/components/ChoiceList/Choice.example.jsx]: Suggest we remove this Right checkboxes with children

File: packages/core/src/components/ChoiceList/Choice.scss [hunk 4]
```diff
@@ -334,21 +340,23 @@ Style guide: components.choice.react
 
 - The `<Choice>` component includes a `checkedChildren` prop that can expose hidden text information or form elements. This **expose within** pattern is especially useful if you need to collect data from follow up questions or give just-in-time feedback.
 - Checked children can be exposed by checking the parent checkbox or radio button
-- Checked children will be announced to screen readers when they are exposed. They have been tested with the following devices:
-  - Windows 10 + Internet Explorer 11 + JAWS screen reader
-  - Windows 10 + Chrome + JAWS
-  - Windows 10 + Firefox + NVDA
-  - MacOS Mojave + Safari + VoiceOver
 - The checkedChildren prop should return one or more items wrapped in a `div` with the following className: `ds-c-choice__checkedChild`. This class sets the spacing and border color for the exposed elements.
```
Comment (bernardwang) [inline packages/core/src/components/ChoiceList/Choice.scss]: Small content nitpick, I noticed other instances of `checkedChildren` are styled as code, should this one be too?

Comment (1Copenut) [reply packages/core/src/components/ChoiceList/Choice.scss]: Yes it should. I'll update now.

**Pull request merged**